### DISCUSSION TITLE
I want to add secret scanning on my own.

### DISCUSSION THREAD

**Discussion by nakatani2023** [aquasecurity/trivy#6166] [Q&A] [answered]
### Question

I want to create custom secret keys in addition to those currently implemented by AWS or Alibaba for secret scanning.
Therefore, I set the types.SecretRuleCategory in trivy/pkg/fanal/secret/builtin-rules.go and added new IDs and Regex to the builtinRules function, but it did not recognize the secret keys.
Is it not possible to add secret keys using this method? I would greatly appreciate it if you could provide guidance on how to implement custom secret keys. Thank you in advance.

### Target

Container Image

### Scanner

Secret

### Output Format

None

### Mode

Standalone

### Operating System

Ubuntu20.04LTS 

### Version

```bash
git clone --depth 1 --branch v0.49.1 https://github.com/aquasecurity/trivy

Version: dev
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-02-20 06:10:27.510746892 +0000 UTC
  NextUpdate: 2024-02-20 12:10:27.510746531 +0000 UTC
  DownloadedAt: 2024-02-20 07:01:53.649798256 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-11-14 00:47:12.884424656 +0000 UTC
  NextUpdate: 2023-11-17 00:47:12.884424406 +0000 UTC
  DownloadedAt: 2023-11-14 08:39:01.516235646 +0000 UTC
```

**Comment by DmitriyLewen** [comment, marked as answer]
Hello @nakatani2023 
Thanks for your interest to Trivy!

You don't need to change code of Trivy.
You can use `trivy-secret.yaml` file with your new secrets - https://aquasecurity.github.io/trivy/v0.49/docs/scanner/secret/#custom-rules.
Also using this file you can enable/disable existing rules.


Regards, Dmitriy

**Reply by DmitriyLewen** [reply]
Keep in mind - we have allow rules (e.g. `example`, `test`, etc.) - https://github.com/aquasecurity/trivy/blob/main/pkg/fanal/secret/builtin-allow-rules.go
Don't forget to disable them when you test your secrets (they may be the reason for the empty result)

**Reply by nakatani2023** [reply]
Hello @DmitriyLewen
Thank you for answer to my question.
I found out that you can add secrets in yaml.

Let me ask you one more question.
It says "Trivy tries to load trivy-secret.yaml in the current directory by default.", but does the current directory mean the directory where you executed the Trivy command? Or do you mean the directory where Trivy is installed?

**Reply by DmitriyLewen** [reply]
> but does the current directory mean the directory where you executed the Trivy command? Or do you mean the directory where Trivy is installed

Directory where Trivy is installed.

But you can set the path using the `--secret-config` flag.

**Comment by nakatani2023** [comment]
I was able to implement the secret key in YAML successfully. Thank you. 
However, the default loading location for YAML was not the installation directory of Trivy, but rather the current directory when executing the Trivy image command. 
For example, if it's "~/test$ trivy image ", then it would be "~/trivy-secret.yaml". 
Since the issue has been resolved, I would like to close this discussion. 
Thank you.

**Reply by DmitriyLewen** [reply]
oh... you are right. Sorry for confusing.

**Reply by nakatani2023** [reply]
Please don't worry about it. Thank you for your kind and helpful response.
